core/backend/src/IpcHost.ts

```diff
@@ -175,27 +175,33 @@ export abstract class IpcHandler {
         return { result: await func.call(impl, ...args) };
       } catch (err: any) {
         let ret: IpcInvokeReturn;
+        let metadata: Exclude<LoggingMetaData, GetMetaDataFunction>;
         if (ITwinError.isITwinError(err)) {
-          const { namespace, errorKey, message, stack, metadata, ...rest } = err;
+          const { namespace, errorKey, message, stack, ...rest } = err;
+          if(rest.metadata)
+            metadata = ITwinError.getMetaData(err);
```

Comment:
as mentioned in https://github.com/iTwin/itwinjs-core/pull/7774#discussion_r1981834324, lets try to keep these apis pure.